### PR TITLE
Update knowledge-distillation.mdx

### PR DESCRIPTION
**Pull request by ghassen-fatnassi**
just some knowledge i learned , though it was good to share , thanks HF

### CODE REVIEW

File: chapters/en/unit3/vision-transformers/knowledge-distillation.mdx [hunk 1]
```diff
@@ -14,29 +12,34 @@ Imagine you were given this multiple-choice question:
...
   <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
 </a>
 
+## So How Can We Use This to Our Advantage?
+
+Knowledge distillation became very useful when the machine learning community realized there was great potential in deploying AI models on edge devices. However, we can't just deploy a model that's 1 GB in size and has a 1-second latency for users to use—it’s too big and too slow. These drawbacks are mainly caused by the large size of the model. From that moment, people started distilling models and reducing their parameter count by more than 90% with negligible performance drops.
+
+### Why Does Knowledge Distillation Work?
+
+1. **Entropy Gain**: Distillation helps in transferring the "dark knowledge" (i.e., soft probabilities) from the teacher model to the student model, which contains more information than hard labels. This extra information helps the student model generalize better.
```
Comment (mmhamdy) [inline chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:41]: What do you mean by "Entropy Gain"?

Comment (mmhamdy) [inline chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:41]:
```suggestion
1. **Entropy Gain**: One way distillation works is by transferring the "dark knowledge" (i.e., soft probabilities) from the teacher model to the student model, which contains more information than hard labels. This extra information helps the student model generalize better.
```

**Review by mmhamdy — COMMENTED** [review]
Thanks for the suggestions! Just a couple of comments.

File: chapters/en/unit3/vision-transformers/knowledge-distillation.mdx (hunk 1 above)
Comment (ghassen-fatnassi) [reply chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:41]: thanks a lot for your fast reply , you are right , entropy gain has no sense , the more convenient term will likely be "information gain" since by absorbing that cross entropy loss between the 2 distributions (student model output & teacher model output) the model gains "more information" than by absorbing cross entropy loss between (student & groundtruth) 
i got confused between the two terms.

File: chapters/en/unit3/vision-transformers/knowledge-distillation.mdx [hunk 2]
```diff
@@ -14,29 +12,34 @@ Imagine you were given this multiple-choice question:
...
 
 To see this loss function implemented in Python and a fully worked out example in Python, let's check out the [notebook for this section](https://github.com/johko/computer-vision-course/blob/main/notebooks/Unit%203%20-%20Vision%20Transformers/KnowledgeDistillation.ipynb).
 
 <a target="_blank" href="https://colab.research.google.com/github/johko/computer-vision-course/blob/main/notebooks/Unit%203%20-%20Vision%20Transformers/KnowledgeDistillation.ipynb">
   <img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/>
 </a>
 
+## So How Can We Use This to Our Advantage?
+
+Knowledge distillation became very useful when the machine learning community realized there was great potential in deploying AI models on edge devices. However, we can't just deploy a model that's 1 GB in size and has a 1-second latency for users to use—it’s too big and too slow. These drawbacks are mainly caused by the large size of the model. From that moment, people started distilling models and reducing their parameter count by more than 90% with negligible performance drops.
```
Comment (johko) [inline chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:37]:
```suggestion
Knowledge distillation became very useful when the machine learning community realized there was great potential in deploying AI models on edge devices. However, we can't just deploy a model that's 1 GB in size and has a 1-second latency for users to use — it’s too big and too slow. These drawbacks are mainly caused by the big number of weights of the model. From that moment, people started distilling models and reducing their parameter count by more than 90% with negligible performance drops.
```

File: chapters/en/unit3/vision-transformers/knowledge-distillation.mdx [hunk 3]
```diff
@@ -14,29 +12,34 @@ Imagine you were given this multiple-choice question:
...
+
+### Why Does Knowledge Distillation Work?
+
+1. **Entropy Gain**: One way distillation works is by transferring the "dark knowledge" (i.e., soft probabilities) from the teacher model to the student model, which contains more information than hard labels. This extra information helps the student model generalize better.
```
Comment (johko) [inline chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:41]: This is a really interesting fact, could you maybe explain soft probabilities a bit more? Our target audience are mainly beginners, so I think it would help to give some more details about that.

File: chapters/en/unit3/vision-transformers/knowledge-distillation.mdx [hunk 4]
```diff
@@ -14,29 +12,34 @@ Imagine you were given this multiple-choice question:
...
+
+2. **Coherent Gradient Updates**: The gradients provided by the teacher model's soft targets help in more stable and coherent updates during training, resulting in a smoother training process for the student model.
```
Comment (johko) [inline chapters/en/unit3/vision-transformers/knowledge-distillation.mdx:43]: Similar to above, maybe you can explain what you mean by soft targets

**Review by johko — CHANGES_REQUESTED** [review]
Thanks for the contribution and I really like the facts you added :+1: 

I left some suggestions to make it even more informative for people going through the course.

**Comment by johko** [comment]
And feel free to add your name in this file in the Vision Transformer part @ghassen-fatnassi: https://github.com/johko/computer-vision-course/blob/main/chapters/en/unit0/welcome/welcome.mdx

**Comment by ghassen-fatnassi** [comment]
done!

**Review by johko — APPROVED** [review]
Thanks for the changes @ghassen-fatnassi , LGTM :+1:

**Review by mmhamdy — COMMENTED** [review]
Hey @ghassen-fatnassi, thanks for this amazing elaboration! 😀 My main issue, as I mentioned previously, is the "entropy gain" section. I think the term is a little bit confusing. In this section, you're explaining what entropy is but it's not clear how this relates to knowledge distillation. I understand you're trying to convey the important role that soft targets play in the student's learning, and how it contain much more information than hard labels, but it would be great to make this connection explicit in this section.

**Review by mmhamdy — APPROVED** [review]
Thanks! Looks good! 👍

**Comment by ghassen-fatnassi** [comment]
there are still merge conflicts to be resolved , idk what should i do as i'm not the code owner and this is still in the back of my head!

**Comment by ghassen-fatnassi** [comment]
what do i do hhhhh *
@mmhamdy
@johko

**Comment by mmhamdy** [comment]
Hey 👋 we have some conflicts in these two files: `welcome.mdx` and `vision-transformer-for-objection-detection.mdx`. can you cancel your changes in these two and create separate PR for them?
